Jenkinsfile

```diff
@@ -86,7 +86,7 @@ for (i = 0; i < buildTypes.size(); i++) {
               if (folders.length > 1) {
                 discoverGitReferenceBuild(scm: folders[1])
               }
-              publishCoverage calculateDiffForChangeRequests: true, adapters: [jacocoAdapter('coverage/target/site/jacoco-aggregate/jacoco.xml')]
+              recordCoverage(tools: [[parser: 'JACOCO', pattern: 'coverage/target/site/jacoco-aggregate/jacoco.xml']], sourceCodeRetention: 'LAST_BUILD')
```

Comment:
https://github.com/jenkins-infra/pipeline-library/pull/649 uses `MODIFIED` rather than `LAST_BUILD`, so I think we should be consistent with that. If `LAST_BUILD` is preferable to `MODIFIED`, then I think it should be changed in both `pipeline-library` _and_ in this repository.

Reply:
Oh, I didn't see the PR. Ulli provided a reasonable explanation why he picked 'MODIFIED'.
Based on that, I shifted our retention too.

Reply:
I used `sourceCodeRetention: 'MODIFIED'`, so we only will see the source code rendering for modified files. Otherwise a lot of unused files will be stored in our workspaces.